### DISCUSSION TITLE
Is there a way to reduce JS bundle size?

### DISCUSSION THREAD

**Discussion by balraj-johal** [pmndrs/react-three-rapier#377] [Q&A]
Hey people,

I'm trying to use react-three-rapier in a project for a client that's targeting a market with typically slow network speeds, and am finding that the gzipped size of the rapier library is having a large impact on the performance of the project.

I was wondering if there's any way I can still use this library while reducing the size of the JS bundle?

Thanks in advance!

**Comment by wiledal** [comment]
Unfortunately not that I am aware. The compiled Rapier library is indeed quite large 😞

**Reply by isaac-mason** [reply]
Related issue: https://github.com/pmndrs/react-three-rapier/issues/115

If we made an alternative build that used `@dimforge/rapier3d` instead of `@dimforge/rapier3d-compat` we could save about 0.5mb.

**Reply by wiledal** [reply]
Could look into this again. When I started work on this I wasn't able to get the non-compat version to compile.

A 0.5 MB save would he huge.

**Reply by balraj-johal** [reply]
I've made an attempt on this, and had decent progress, but I seem to be at a roadblock getting the wasm plugin working with vitest. I've detailed what I did and what the issue is in #115 if that is of any use :)
